File: lect2/codes/decl_functions2.py
import math
def taylorSin(at, maxtoler, maxiter):
    result , real_value = 0.0, math.sin(at)
    for i in range(0, maxiter):
        den = math.factorial(2 * i + 1)
        num, mult = (-1) ** i, at ** (2 * i + 1)
        result += (num / den) * mult
        if abs(real_value - result) < maxtoler:
            break
    return result
if __name__ == "__main__":
    print(taylorSin(7/8, 10**-8, 100))

File: lect2/codes/decl_functions2.m
clc;
function [res err] = taylorSin(x, niter);
  result = 0.0;
  for i=0:niter;
    den = factorial(2 * i + 1);
    num = (-1) ^ i;
    mult = (x) ^ (2 * i + 1);
    result = result + (num / den) * mult;
  endfor
  res = result;
end
disp(taylorSin(1/2, 100));